pyprof/nvtx/nvmarker.py

```diff
@@ -68,58 +70,141 @@ def isfunc(mod, f):
     return ins.ismethod(attr) or ins.isfunction(attr) or ins.ismethoddescriptor(attr) or ins.isbuiltin(attr)
 
 
+# Nested dicts of this run's frame names to help uniquify them
+#
+# func_map[(partial_func_stack,frame_name)][filename+lineno] = frame_name_to_use
+#
+func_map = {}
+
 # Returns a dict string with a tracemarker and function stack in it
 #
-def traceMarker():
+def traceMarker(op_name):
+    global func_map
 
     # Return True if the name in the hierarchy should be skipped
-    def should_skip_frame_name(name):
-        if name in ["__call__","wrapper_func","<module>", "always_benchmark_wrapper"]:
+    #
+    def should_skip_frame_name(name, prev_name):
+        # __call__:  
+        #    Much of Torch library is implemented in this way. Ignore these extra layers
+        # wrapper_func and always_benchmark_warpper: 
+        #    Are functions in this file. If there are nested monkeypatched functions 
+        #    we don't want it to show up
+        # <*>: 
+        #    Things like <module>, <genexpr>, <lamba> which don't add any information 
+        #    and break html
+        #
+        for prefix in ["__call__", "wrapper_func", "always_benchmark_wrapper"]:
+            if name.startswith(prefix):
+                return True
+        if name.startswith("<") and name.endswith(">"):
             return True
-        elif name.startswith("<") and name.endswith(">"):
+        if name==prev_name:
```

Comment:
Can we comment this one as well?
Something like `Removing duplicate names in hierarchy.`

Reply:
> 
> 
> Couple small changes.
> Have you run a small profile via pyprof to ensure this doesn't break profiling and reporting in pyprof?

I ran the lenet CI, which is failing because it was already failing.

Reply:
done